nomos-cli/src/cmds/executor/mod.rs

```diff
@@ -0,0 +1,140 @@
+// std
+use std::path::PathBuf;
+// crates
+use clap::Args;
+use reqwest::Url;
+// internal
+use executor_http_client::ExecutorHttpClient;
+use kzgrs_backend::{
+    common::blob::DaBlob,
+    dispersal::{Index, Metadata},
+};
+
+#[derive(Args, Debug)]
+pub struct Disseminate {
+    /// Text to disseminate.
+    #[clap(short, long, required_unless_present("file"))]
+    pub data: Option<String>,
+    /// File to disseminate.
+    #[clap(short, long)]
+    pub file: Option<PathBuf>,
+    /// Application ID for dispersed data.
+    #[clap(long)]
+    pub app_id: String,
+    /// Index for the Blob associated with Application ID.
+    #[clap(long)]
+    pub index: u64,
+    /// Executor address which is responsible for dissemination.
+    #[clap(long)]
+    pub addr: Url,
+}
+
+impl Disseminate {
+    pub fn run(&self) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Why by ref?

Reply:
Fixed, as we are not reusing chaining commands it's not necessary.